hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/configuration/OptionsResolver.java

```diff
@@ -86,6 +87,15 @@ public static boolean supportRowDataAppend(Configuration conf, RowType rowType)
             || WriteOperationType.valueOf(conf.get(FlinkOptions.OPERATION).toUpperCase()) == WriteOperationType.DELETE);
   }
 
+  /**
+   * Returns whether current index is partition level simple bucket index based on given configuration {@code conf}.
+   */
+  public static Boolean isPartitionLevelSimpleBucketIndex(Configuration conf) {
+    HoodieIndex.BucketIndexEngineType engineType = OptionsResolver.getBucketEngineType(conf);
+    return engineType.equals(HoodieIndex.BucketIndexEngineType.SIMPLE)
```

Comment:
I didn't see the fix on reader side, for e.g. the bucket index bucket pruning, does the query still work if the pruning happens?

Reply:
The partition-level BucketIndex adaptation for Flink and Spark's BucketID pruning would require approximately 1500+ additional lines of code, which was not implemented in this PR. Perhaps we can split the Partition Bucket Index implementation into Writer and Reader components. In this PR, we focus on the Writer implementation details, while temporarily adding a protective mechanism in the Reader portion.

Reply:
>  Perhaps we can split the Partition Bucket Index implementation into Writer and Reader component

if the reader fix also aims to be included in release 1.1, I'm ok with it.

Reply:
Sure, will get it done ASAP. Create https://issues.apache.org/jira/browse/HUDI-9230 to track it